embark-ui/src/containers/FileExplorerRowContainer.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import {UncontrolledTooltip} from 'reactstrap';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import FontAwesome from 'react-fontawesome';
+
+import {removeFile as removeFileAction, saveFile as saveFileAction, saveFolder as saveFolderAction} from '../actions';
+import AddFileModal from '../components/AddFileModal';
+import AddFolderModal from '../components/AddFolderModal';
+
+class FileExplorerRowContainer extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {active: false};
+    this.addFileModal = React.createRef();
+    this.addFolderModal = React.createRef();
+  }
+
+  activateNode() {
+    this.setState({active : true});
+  }
+
+  deactivateNode() {
+    this.setState({active : false});
+  }
+
+  renderAction() {
+    return (
+      <span className="float-right mr-2">
+        {this.props.node.children &&
+          <React.Fragment>
+            <a id="add-file" onClick={() => this.addFileModal.current.toggle()}><FontAwesome name="plus" className="text-success mr-2" /></a>
```

Comment:
Since this is an `<a>`, shouldn't you `preventDefault()` on the event so that we don't mess up the url or something? Also, isn't `href` an obligatory attribute?
Actually, it might be better to just replace `<a>` by a `<span>` or something with a `cursor: pointer` css property.

Reply:
Same for the other 2 below